src/stochastic_search_iterator.jl

```diff
@@ -0,0 +1,174 @@
+costs that are worse. 
+`cost` means how different are the outcomes of the program compared to the correct outcomes.
+The lower the `cost` the better the program performs on the examples. The `cost` is provided by the `cost_function`
+
+    accept(current_cost::Float, possible_cost::Float, temperature::Float) -> Bool
+----
+Returns the new temperature based on the previous temperature. Higher the `temperature` means that the algorithm will explore more.
+    
+    temperature(previous_temperature::Float) -> Float 
+---
+Returns the cost of the current program. It receives a list of tuples `(expected, found)` and gives back a cost.
+    
+    cost_function(outcomes::Tuple{Int64,Int64}[]) -> Float
+
+----
+# Fields
+-   `grammar::ContextFreeGrammar` grammar that the algorithm uses
+-   `max_depth::Int64 = 5`  maximum depth of the program to generate
+-   `examples::Vector{Example}` example used to check the program
+-   `neighbourhood::Function` 
+-   `propose::Function`
+-   `accept::Function`
+-   `temperature::Function`
+-   `cost_function::Function`
+-   `start_symbol::Symbol` the start symbol of the algorithm `:Real` or `:Int`
+-   `initial_temperature::Int64` = 1 
+An iterator over all possible expressions of a grammar up to max_depth with start symbol sym.
+"""
+Base.@kwdef mutable struct StochasticSearchEnumerator <: ExpressionIterator
+    grammar::ContextFreeGrammar
+    max_depth::Int64 = 5  # maximum depth of the program that is generated
+    examples::Vector{Example}
+    neighbourhood::Function
+    propose::Function
+    accept::Function
+    temperature::Function
+    cost_function::Function
+    start_symbol::Symbol
+    initial_temperature::Int64 = 1
+end
+
+Base.@kwdef struct IteratorState
+    current_program::RuleNode
+    current_temperature::Float32
+    best_program::RuleNode
+    best_program_cost::Float32
+end
+
+Base.IteratorSize(::StochasticSearchEnumerator) = Base.SizeUnknown()
+Base.eltype(::StochasticSearchEnumerator) = RuleNode
+
+function Base.iterate(iter::StochasticSearchEnumerator)
+    grammar, max_depth = iter.grammar, iter.max_depth
+    # sample a random node using start symbol and grammar
+    sampled_program = rand(RuleNode, grammar, iter.start_symbol, max_depth)
+    current_cost = calculate_cost(sampled_program, iter.cost_function, iter.examples, iter.grammar)
+    return (sampled_program, IteratorState(
+        current_program=sampled_program,
+        current_temperature=iter.initial_temperature,
+        best_program=sampled_program,
+        best_program_cost=current_cost))
+end
+
+
+"""
+The algorithm that constructs the iterator of StochasticSearchEnumerator. It has the following structure:
+
+1. calculate the cost of the current program
+1. get a random node location -> location,dict = neighbourhood(current_program)
+2. call propose on the current program 
+"""
```

Comment:
There are two 1s in the docstring

Reply:
We updated the documentation. See [here](https://github.com/Herb-AI/HerbSearch.jl/blob/143921ffbea68b955f634b6470c921ed44f4a70d/src/stochastic_search_iterator.jl#LL90C1-L90C1)
```
The algorithm that constructs the iterator of StochasticSearchEnumerator. It has the following structure:

1. get a random node location -> location,dict = neighbourhood(current_program)
2. call propose on the current program getting a list of possbile replacements in the node location 
3. iterate through all the possible replacements and perform the replacement in the current program 
    4.  accept the new program by modifying the next_program or reject the new program
5. return the new next_program
```